File: short_amount.cpp
// https://programmers.co.kr/learn/courses/30/lessons/82612

using namespace std;

long long solution(int price, int money, int count)
{
    long long sum = 0;
    for (long long i = 1; i <= count; i++)
    {
        sum += i * price;
    }
    if (sum > money)
        return sum - money;
    else
        return 0;
}

// typedef long long ll;

// long long solution(int price, int money, int count)
// {
//     ll answer = (ll)(count * (count+1) / 2) * price;
//     return answer > money ? answer - money : 0;
// }

File: python/short_amount.py
# https://programmers.co.kr/learn/courses/30/lessons/82612


def solution(price, money, count):
    answer = 0
    sum = 0
    for i in range(1, count + 1):
        sum += i * price
    if sum > money:
        answer = sum - money

    return answer


# def solution(price, money, count):
#     return max(0,price*(count+1)*count//2-money)
